iob.bat

```diff
@@ -17,10 +17,18 @@ if [%1]==[fix] (
                     node node_modules/iobroker.js-controller/iobroker.js %1 %2 %3 %4 %5 %6 %7 %8
                 )
             ) else (
-                node node_modules/iobroker.js-controller/iobroker.js %1 %2 %3 %4 %5 %6 %7 %8
+				if [%1]==[restart] (
+					if [%2]==[] (
+						call serviceIoBroker.bat restart
+					) else (
+						node node_modules/iobroker.js-controller/iobroker.js %1 %2 %3 %4 %5 %6 %7 %8
```

Comment:
```suggestion
						node node_modules/iobroker.js-controller/iobroker.js %*
```

Not related to this PR, however if we are touching it now. And on all other occurrences, I personally find it suboptimal to pass only a fixed number of args, at some point some would may run into strange hard to debug issues.

Nitpick (and probably personal pref): Also if `iobroker.bat` is the original file and `iob.bat` acts as some kind of alias, I would keep the real code in `iobroker.bat` and do `call iobroker.bat %*` in `iob.bat` 

However, as @Apollon77 and you are right now into the topic, any chance to get rid of one of these two files?

Reply:
ioBroker.bat causes trouble because npx calls like ...
`@iobroker/fix`
invoke ioBroker.bat instead of invoking the npm package.
AFAIK @GermanBluefox decided to remove ioBroker.bat as soon as possible.
I'm personally with replacing all '%1 %2 %3 %4 %5 %6 %7 %8' with %*

Reply:
Thanks, Ingo explained to me in detail now. We currently have some duplicate code, because if installer is not used, controller also has to provide the `iob.bat`. As installer also installs controller from my pov it should be enough to provide the file in js-controller instead of having some redundancy which gets out of sync real quick and is also currently out of sync. Any arguments against it?

Reply:
Is there a possibility to install the controller under Windows without the "Installer"? "Installer" means calling
`npx @iobroker/install`
Sorry, ich muss zwecks besserer Verstädigung auf deutsch wechseln:
Ich habe immer noch nicht kapiert, wer das iob.bat, das beim Controller abliegt, verwendet. Ich kenne derzeit kein Szenario, wo dieses File benötigt wird. Evtl. BackItUp? Für den Anwender müssen die Dateien letztendlich im Root Verzeichnis der ioBroker "Installation" landen, sonst sieht er die ja gar nicht.
Bitte beachten, dass in den Batch Files, die bei der Windows-Installation mitkommen, Änderungen vorgenommen wurden, um alternative Service Namen zu unterstützen.
Evtl. reden wir auch aneinander vorbei. Wir können gern mal telefonieren bzw. auf Discord zum Gespräch verabreden. Evtl. macht es das einfacher. :-)

Reply:
Ja, ist möglich war wohl früher mal die Vorgehensweise lt. Ingo: Directory manuell erstellen, dann Controller via npm installieren. Wenn Änderungen vorgenommen werden, wäre ich wieder eher bei das File im Controller entfernen und die Nutzung anpassen. Und sagen ab Controller v5 darf man nur noch mit der neuen Methode installieren. 

So wie ich das sehe nutzt mindestens der Controller selbst die `restart.js` welche die bat im aktuellen Dir möchte. 

https://github.com/ioBroker/ioBroker.js-controller/blob/06227a927134448afdbb30908b1df8cae3017b76/packages/controller/src/lib/restart.ts#L15

Wäre allerdings dann die einzige Stelle die anzupassen ist. @Apollon77 Meinungen?

Reply:
Die Frage wäre, wann restart.js aufgerufen wird. Der Restart Button im Admin führt jedenfalls bereits zu ioBroker.bat im Root Verzeichnis (also das, das der "Installer" liefert. Und ja, ich wäre dafür, die Bat Files aus dem Controller zu entfernen, wenn es keinen guten Grund gibt, sie zu behalten. S. auch mein Issue bein Controller:
https://github.com/ioBroker/ioBroker.js-controller/issues/2219

Reply:
> Ja, ist möglich war wohl früher mal die Vorgehensweise lt. Ingo: Directory manuell erstellen, dann Controller via npm installieren.
Soweit ich das sehe, wird in diesem Fall kein Windows Service angelegt, oder?

Reply:
Korrekt und allgemein finde ich diese Installationen erhole unschick und der installer ist zu bevorzugen. Daher wäre ich da für deprecaten. 

Hast du Telegram? Ich habe noch eine Frage zum Update Prozess unter Windows An dich als Experten. Da wir gerne backward kompatibel bleiben möchten.